### ISSUE TITLE
GitHub services deprecation, replacements needed

### ISSUE THREAD

**Issue by DrMcCoy** [xoreos/xoreos#326]
GitHub has deprecated their "services" a while ago, to be removed on **January 31, 2019**.

The "services" are the things that listen for repository pushes, incoming pull requests, etc. and announce them a multitude of channels.

What will stay, and what is supposed to be the replacement, are webhooks. Which is means the only thing GitHub will do is notify an URL via http(s). So that means we (or I, probably, on the xoreos.org server) will have to set up something that listens on http(s) on a server, and then itself knows how to communicate to the thing we want the announcement to go.
For us, these are the three we're using:
* Travis CI
* Email
* IRC

The first one, Travis CI, is probably not our problem. I'm assuming Travis CI will have something ready we can just plug into the webhooks. I'll be ignoring it for now.

Email here only means the push announcements which go to the [xoreos-git](https://xoreos.org/mailman/listinfo/xoreos-git) mailing list, not the PR mails (because they come from GitHub directly, when you watch a repository/PR/...). If it comes to that, I'm okay with losing xoreos-git mails. But if we're going to replace this, I'd like to improve them, by having the mail include the full diff, if possible.

That last one, IRC, is the one I'd *really* miss. Currently, the bot is already shut out (*), and I'm not happy with that.

So what we need is one (or two separate) service I can run on the xoreos.org service that listens to http(s) requests from GitHub and can announce it via email and IRC. Ideally, not written in a "weird" language and without a whole host of dependencies. Easy to maintain. Already in use somewhere, so we know it's not going to break every week. And a pony.

For the email thing, I'd like to have it include diffs if possible (see above), and the IRC thing would need to identify with Freenode's NickServ (see below).

If anybody knows of something like that, please let me know.

To recap, what we need:
- A service that listens on http(s) for GitHub to poke it on a push, PR, ...
- Can connect to IRC to announce that in #xoreos on Freenode
  - Can identify with NickServ before joining #xoreos
- Can send emails to announce pushes/commits
  - Include full diff of the commits
- Easy to set up
  - With documentation
- Easy to maintain
- Secure, established

(*) Due to the ongoing spam attack on Freenode, I made the channel +r, meaning only registered users can join. The GitHub IRC service can theoretically identify with Freenode's NickServ, but it's not working for some reason. I can't really debug it, either. What I'm currently *assuming* is that there's a race condition, with the GitHub bot seeing the initial "This nickname is registered, please identify" message as the reply to the identify command. But that's just a guess.

**Comment by clone2727** [comment]
I started on a bot in Python (Sopel) for #xoreos last year. I can pick up on that again. As long as you have somewhere to run it, that is.

Is it just a REST API to retrieve recent commits over https?

**Comment by DrMcCoy** [comment]
> I started on a bot in Python (Sopel) for #xoreos last year. I can pick up on that again.

That would be really great. The timeline is *kinda* pressing, though, that's why I had hoped for something existing.

> As long as you have somewhere to run it, that is.

I should be able to run it on the xoreos.org server, yes.

> Is it just a REST API to retrieve recent commits over https?

I think so, yes. It's probably not great if it just polls for changes constantly, but also listens for GitHub to contact it via the webhooks thing. So it would need to listen on http(s) too.

**Comment by clone2727** [comment]
Sopel does most of the IRC heavy lifting for us; I just have to hook into it with some plugins.

I can just have it set up an HTTPS listener and receive the POST events.

Definitely should be able to get it done by January :)

**Comment by lynxlynxlynx** [comment]
We're in the same boat. While searching, I found two things:
- the github closed beta of actions (or something like this) can solve it
- pma's script to be ran on a server
https://github.com/williamdes/pma-scripts/commit/9d2e9ac48698a5f24cdc1d3735e2624f55f7489a
- https://app.github-file-watcher.com/ but not tried, so no idea what is the content that it sends

**Comment by DrMcCoy** [comment]
Yeah, I heard that maybe GitHub Actions might be able to do that. Though information on how exactly GitHub Actions work are sparse. I opened an "investigate" issue some time ago, because they might be useful for other things too: https://github.com/xoreos/xoreos/issues/390

I still don't like that they're switching the services off at all :P

**Comment by lynxlynxlynx** [comment]
Actually it does work, just with a delay and some aggregation. For example:

> Subject: [GitHub File Watcher] New changes in gemrb/gemrb
> Date: Fri, 11 Jan 2019 15:54:40 +0000
> 
> There are new matches in your subscription to changes in **gemrb/gemrb** to files that match
> *****!
> 
> Here's what changed:
> 
> 
>   + [admin/run.cachegrind](https://github.com/gemrb/gemrb/blob/b0bf24e25024cdfd37cc45e650c2aa3135dc9677/admin/run.cachegrind) (added)
> 
>   + [gemrb/core/Inventory.cpp](https://github.com/gemrb/gemrb/blob/b0bf24e25024cdfd37cc45e650c2aa3135dc9677/gemrb/core/Inventory.cpp) (modified)
> 
> 
> You can [view those changes on GitHub](https://github.com/gemrb/gemrb/compare/gemrb:be6fa64...gemrb:b0bf24e).
> 
> ---
> 
> Visit GitHub File Watcher to [manage this watch or unsubscribe]().
> 
> -- Thomas, GitHub File Watcher

Sadly no diffs, but still better than nothing.

**Comment by lynxlynxlynx** [comment]
Well, I don't know whether to be annoyed or happy, but the deadline is passed and while the warning did mention they'll find a replacement ... nothing was communicated.

However we just keep getting the emails in February without the depreciation warning now, so I guess it's a non-issue? Weirdos.

Found this in the settings:
> The Email service has been replaced by repository notifications. Settings have been migrated on your behalf and can be configured there. 

I guess you can close this. :)

**Comment by DrMcCoy** [comment]
https://xoreos.org/pipermail/xoreos-devel/2019-February/000084.html

**Comment by HumanG33k** [comment]
Maybe too late but i m pretty sure hooks can be propagated from travis to irc.
